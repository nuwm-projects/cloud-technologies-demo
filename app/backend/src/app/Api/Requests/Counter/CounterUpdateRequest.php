<?php

namespace App\Api\Requests\Counter;

use Illuminate\Foundation\Http\FormRequest;

class CounterUpdateRequest extends FormRequest
{
    /**
     * @return string[]
     */
    public function rules(): array
    {
        return [
            'count' => 'required|numeric',
        ];
    }
}
