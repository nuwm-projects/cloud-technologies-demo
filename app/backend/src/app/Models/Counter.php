<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
 * @property $count
 */
class Counter extends Model
{
    protected $attributes = [
        'count' => 0,
    ];

    /**
     * @param int $number
     */
    public function addNumber(int $number): void
    {
        if ($number < 0) {
            $number = 0;
        }

        $this->count += $number;
    }
}
