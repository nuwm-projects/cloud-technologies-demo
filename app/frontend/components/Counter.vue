<template>
    <div class="d-flex justify-content-center">
        <b-card
            title="Counter"
            tag="article"
            style="max-width: 20rem;"
            class="mb-2"
        >

            <b-card-text>
                {{ counter }}
            </b-card-text>

            <b-button href="#" variant="primary" @click="plus">Plus</b-button>
        </b-card>
    </div>
</template>

<script>
export default {
    name: 'Counter',
    data() {
        return {
            counter: 0,
        }
    },
    mounted() {
        this.$axios.get('/counter')
            .then((response) => {
                this.counter = response.counter;
            });
    },
    methods: {
        async plus() {
            this.counter++;

             await this.$axios.put('/counter', {
                counter: this.counter,
            })
        },
    }
}
</script>
