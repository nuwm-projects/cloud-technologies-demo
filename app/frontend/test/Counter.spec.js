import {describe, expect, jest, test} from "@jest/globals";
import {mount} from "@vue/test-utils";
import Counter from "@/components/Counter";
import Vue from "vue";
import {BootstrapVue} from 'bootstrap-vue'

Vue.use(BootstrapVue);

describe('Counter', () => {
    let wrapper;

    const createComponent = (config) => {
        wrapper = mount(Counter, config);
    }


    test('Test rendered component', () => {
        createComponent({
            mocks: {
                $axios: {
                    get: jest.fn(() => Promise.resolve({data: {count: 1}}))
                }
            }
        });

        expect(wrapper).toMatchSnapshot();
    });

    test('Press increment', async () => {
        createComponent({
            mocks: {
                $axios: {
                    get: jest.fn(() => Promise.resolve({data: {count: 1}})),
                    post: jest.fn(),
                }
            }
        });

        await wrapper.find('a[role="button"]').trigger('click');
        await wrapper.vm.$nextTick();

        expect(wrapper.find('.counter-number').text()).toBe('1');
    });
});
