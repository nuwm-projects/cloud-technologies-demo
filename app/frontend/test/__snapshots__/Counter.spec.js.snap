// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Test rendered component 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
